### PR TITLE
Implement saved scenario deletion

### LINKED ISSUE #3523: Add possibility to delete/archive scenarios
1. I do make a lot of scenario's. Also with version's of them. So my dashboard and all scenario's got a little diffused and cluttered. 

2. Is it possible to delete a scenario? (or more at once...). 

3. If so, how? 

4. If not, is this an option to build in?

### PR DESCRIPTION
**Pull request by antw**
- Saved scenarios can now be deleted by users.

- When a scenario is deleted, either by clicking the "Delete" button in the scenario list, or selecting "Settings -> Move to Trash" on the scenario page, it is sent to the trash.

- Scenarios in the trash may be restored within 60 days of being placed there, after which they'll be deleted permanently.

- A user may opt to permanently delete a scenario immediately from the trash.

- Admin users will no longer see everyone's saved scenarios.

Closes #3523

### Scenario list

Screenshot 2022-05-18 at 16 28 23

Screenshot 2022-05-18 at 16 28 55

### Trash empty state

Screenshot 2022-05-18 at 16 31 09

### Non-empty trash

Screenshot 2022-05-18 at 16 28 33

### Saved scenario

Screenshot 2022-05-18 at 16 34 01

/cc @ChaelKruip

### CODE REVIEW

**Comment by antw** [comment]
Forgot to mention this in the main post, there are a couple of translations needed, please. 🙏

* [These please](https://github.com/quintel/etmodel/blob/64356e0871ab525b084cf3b8643c46e45f21c47a/config/locales/nl_etm.yml#L253-L262).
* [These were courtesy of Google Translate, but should probably be double checked](https://github.com/quintel/etmodel/blob/64356e0871ab525b084cf3b8643c46e45f21c47a/config/locales/nl_etm.yml#L30-L35).

**Comment by marliekeverweij** [comment]
Wow this is so cool!!!!

**Comment by ChaelKruip** [comment]
Very nice!

**Comment by noracato** [comment]
Very nice, thanks for implemeting this! The only thing is that locally for me it doesn't look as nice as in your screenshots. As if there's some style info missing. Should I install something extra? I already tried a `yarn install`. 

Screenshot 2022-05-20 at 13 21 47

> there are a couple of translations needed

Done 😊

**Comment by antw** [comment]
> The only thing is that locally for me it doesn't look as nice as in your screenshots. As if there's some style info missing

It looks as though your local Rails app is serving stale assets. Try running `bundle exec rails assets:clobber` to get rid of any precompiled assets and restart the app. Hopefully, that will fix it.

**Comment by noracato** [comment]
> Hopefully, that will fix it.

It did, thanks!

File: app/controllers/saved_scenarios_controller.rb [hunk 1]
```diff
@@ -114,10 +126,53 @@ def update
...
 
+  # Soft-deletes the scenario so that it no longer appears in listings.
+  #
+  # PUT /saved_scenarios/:id/discard
+  def discard
+    unless @saved_scenario.discarded?
+      @saved_scenario.discarded_at = Time.zone.now
+      @saved_scenario.save(touch: false)
+
+      flash.notice = 'Your scenario was put in the trash'
```
Comment (noracato) [inline app/controllers/saved_scenarios_controller.rb:137]: Would you mind if we put up some translations for the notices as well? Having it in English here in between all the Dutch makes it look somehow unfinished to me. If you like the idea I can provide the translations 🙂 

Screenshot 2022-05-20 at 15 57 46

**Review by noracato — APPROVED** [review]
Nice work! I like what you did with the design and the new trash tab, and it works nicely. A longtime wish fulfilled! 🎉 

I made one comment, we can see if we want to act upon it.

File: app/controllers/saved_scenarios_controller.rb (hunk 1 above)
Comment (antw) [reply app/controllers/saved_scenarios_controller.rb]: I created the first version in English only and thought I had caught all these. Apparently not!

I moved them into the locale files. There are [two more messages to be translated](https://github.com/quintel/etmodel/blob/bfd847da4669a2414291a37f049de2785625a1a3/config/locales/nl_etm.yml#L257-L258), please.

**Pull request merged**